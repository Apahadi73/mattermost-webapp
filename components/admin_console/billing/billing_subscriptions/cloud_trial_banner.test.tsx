// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {shallow} from 'enzyme';
import expect from 'expect';

import CloudTrialBanner from './cloud_trial_banner';

jest.mock('components/common/hooks/useOpenSalesLink', () => ({
    __esModule: true,
    default: () => () => true,
}));

describe('components/admin_console/billing_subscription/CloudTrialBanner', () => {
    test('should match snapshot when trial end date is 0', () => {
        const wrapper = shallow(<CloudTrialBanner trialEndDate={0}/>);
        expect(wrapper).toMatchSnapshot();
    });

    test('should match snapshot when is cloud and is free trial', () => {
        const wrapper = shallow(<CloudTrialBanner trialEndDate={12345}/>);
        expect(wrapper).toMatchSnapshot();
    });
});
