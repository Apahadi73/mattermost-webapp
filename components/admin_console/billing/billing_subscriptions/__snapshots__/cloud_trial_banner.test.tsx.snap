// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/admin_console/billing_subscription/CloudTrialBanner should match snapshot when is cloud and is free trial 1`] = `
<AlertBanner
  actionButtonLeft={
    <UpgradeLink
      buttonText="Upgrade"
      styleButton={true}
    />
  }
  actionButtonRight={
    <button
      className="AlertBanner__buttonRight"
      onClick={[Function]}
    >
      <Memo(MemoizedFormattedMessage)
        defaultMessage="Contact sales"
        id="admin.billing.subscription.privateCloudCard.contactSalesy"
      />
    </button>
  }
  hideIcon={true}
  message={
    <Memo(MemoizedFormattedMessage)
      defaultMessage="Your trial ends on {date} {time}. Upgrade to one of our paid plans with no limits."
      id="admin.subscription.cloudTrialCard.description"
      values={
        Object {
          "date": "Jan 1, 1970 ",
          "time": "11:59 pm GMT",
        }
      }
    />
  }
  mode="info"
  title={
    <Memo(MemoizedFormattedMessage)
      defaultMessage="Upgrade to one of our paid plans to avoid Starter plan data limits"
      id="admin.subscription.cloudTrialCard.upgradeTitle"
    />
  }
/>
`;

exports[`components/admin_console/billing_subscription/CloudTrialBanner should match snapshot when trial end date is 0 1`] = `""`;
